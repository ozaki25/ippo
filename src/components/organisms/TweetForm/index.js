import React from 'react';
import styled from 'styled-components';
import { Text } from '@blueprintjs/core';
import propTypes from 'prop-types';
import { Button, FormGroup, TextArea } from '@blueprintjs/core';

const Error = styled(Text)`
  color: red;
`;

class TweetForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: `
#${this.props.hashtag}`,
      error: false,
    };
    this.text = React.createRef();
    console.log(props);
  }

  componentDidMount() {
    this.text.current.focus();
    this.text.current.setSelectionRange(0, 0);
  }

  onClick = async () => {
    if (!this.invalid()) {
      const { text } = this.state;
      const { hashtag, createTweet, onFinished } = this.props;
      const tweet = {
        text,
        hashtag,
        name: 'ozaki25', // TODO
        time: new Date().toString(),
      };
      const result = await createTweet({
        variables: { tweet },
      }).catch(e => console.log(e));
      console.log(result);
      onFinished();
    } else {
      this.setState({ error: true });
    }
  };

  onChange = event => this.setState({ [event.target.name]: event.target.value });

  invalid = () => this.state.text.trim() === '';

  render() {
    return (
      <form>
        {this.state.error && <Error>入力内容を確認して下さい</Error>}
        <FormGroup labelFor="text">
          <TextArea
            id="text"
            name="text"
            onChange={this.onChange}
            value={this.state.text}
            inputRef={this.text}
            large
            fill
          />
        </FormGroup>
        <Button onClick={this.onClick} disabled={this.invalid()} text="ツイート" fill large />
      </form>
    );
  }
}

TweetForm.displayName = 'TweetForm';

TweetForm.propTypes = {
  createTweet: propTypes.func.isRequired,
  hashtag: propTypes.string.isRequired,
  onFinished: propTypes.func.isRequired,
};

TweetForm.defaultProps = {};

export default TweetForm;
