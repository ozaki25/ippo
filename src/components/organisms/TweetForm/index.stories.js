import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import TweetForm from '.';

const stories = storiesOf('organisms/TweetForm', module);

stories.add('通常パターン場合', () => <TweetForm onSubmit={action('onSubmit')} />);
