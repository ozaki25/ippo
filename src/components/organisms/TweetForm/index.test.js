import React from 'react';
import { snapshot } from 'test/helpers';
import TweetForm from '.';

const props = {
  onSubmit: jest.fn(),
};

snapshot('TweetForm/nomal', <TweetForm {...props} />);
