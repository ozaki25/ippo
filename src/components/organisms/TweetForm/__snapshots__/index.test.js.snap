// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: TweetForm/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <TweetForm
    onSubmit={[MockFunction]}
  >
    <form>
      <Blueprint3.FormGroup
        labelFor="text"
      >
        <div
          className="bp3-form-group"
        >
          <div
            className="bp3-form-content"
          >
            <Blueprint3.TextArea
              fill={true}
              id="text"
              inputRef={
                Object {
                  "current": <textarea
                    class="bp3-input bp3-fill bp3-large"
                    id="text"
                    name="text"
                  >
                    
#undefined
                  </textarea>,
                }
              }
              large={true}
              name="text"
              onChange={[Function]}
              value="
#undefined"
            >
              <textarea
                className="bp3-input bp3-fill bp3-large"
                id="text"
                name="text"
                onChange={[Function]}
                value="
#undefined"
              />
            </Blueprint3.TextArea>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.Button
        disabled={false}
        fill={true}
        large={true}
        onClick={[Function]}
        text="ツイート"
      >
        <button
          className="bp3-button bp3-fill bp3-large"
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          type="button"
        >
          <Blueprint3.Icon
            key="leftIcon"
          />
          <span
            className="bp3-button-text"
            key="text"
          >
            ツイート
          </span>
          <Blueprint3.Icon
            key="rightIcon"
          />
        </button>
      </Blueprint3.Button>
    </form>
  </TweetForm>
</MockRouter>
`;
